experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts

```diff
@@ -29,7 +29,7 @@ export abstract class InstrumentationBase<T = any>
   extends InstrumentationAbstract
   implements types.Instrumentation {
   private _modules: InstrumentationModuleDefinition<T>[];
-  private _hooks: RequireInTheMiddle.Hooked[] = [];
```

Comment:
please keep this an array, we wanted to keep this around if we needed it.

Reply:
What would we put in the array? There is only one hook now.

Reply:
I agree that I'm not sure I see the value in storing the array reference. We aren't using it for anything and it is just memory kept around for little benefit. We could keep a reference to either the singleton or to the RITM hook itself if we needed it for testing instead of just a boolean?

Reply:
> What would we put in the array? There is only one hook now.

Depending on the instrumentation we might have multiple hooks for reach module patched, also with ESM we'll need to store two ref for each modules (https://github.com/open-telemetry/opentelemetry-js/pull/2846)

> I agree that I'm not sure I see the value in storing the array reference. We aren't using it for anything and it is just memory kept around for little benefit

It isn't used right now so you are right that there is little benefit but i expect we might want at some point to cleanup hooks when instrumentation is disabled (which we should do anyway). I'm not stronly against it but it might be a nice feature in the future to have access to those refs. Nothing that the memory overhead is quite low since we are only stored 1-4 refs inside an array, refs object aren't necesarelly there anyway

Reply:
That makes sense. I will address it.